In [1]:
import pandas as pd


#At first I will take file as wine_data(it is just a name, you can take every name you want and we are using function pandas it is for analyzing the data and in order to use short version of pandas we import pandas as pd 

In [2]:
wine_data = pd.read_csv("bank.csv", sep=";")


#The data is related with direct marketing campaigns of a Portuguese banking institution. The marketing campaigns were based on phone calls. Often, more than one contact to the same client was required, in order to access if the product (bank term deposit) would be (or not) subscribed. 
#The file we are using bank.csv is consist of list of people and 17 features(things that espessially of a person or thing, in our case person) that define every person individually. The goal is to do correaltion(statistical relationship of one thing with another thing, weather causal or not) of this file. P.S. Mainly is about deposit,loan in the bank. 

In [8]:
wine_data

age            job  marital  education default  balance housing loan  \
0      30     unemployed  married    primary      no     1787      no   no   
1      33       services  married  secondary      no     4789     yes  yes   
2      35     management   single   tertiary      no     1350     yes   no   
3      30     management  married   tertiary      no     1476     yes  yes   
4      59    blue-collar  married  secondary      no        0     yes   no   
...   ...            ...      ...        ...     ...      ...     ...  ...   
4516   33       services  married  secondary      no     -333     yes   no   
4517   57  self-employed  married   tertiary     yes    -3313     yes  yes   
4518   57     technician  married  secondary      no      295      no   no   
4519   28    blue-collar  married  secondary      no     1137      no   no   
4520   44   entrepreneur   single   tertiary      no     1136     yes  yes   

       contact  day month  duration  campaign  pdays  previous poutcome   y  
0     cellular   19   oct        79         1     -1         0  unknown  no  
1     cellular   11   may       220         1    339         4  failure  no  
2     cellular   16   apr       185         1    330         1  failure  no  
3      unknown    3   jun       199         4     -1         0  unknown  no  
4      unknown    5   may       226         1     -1         0  unknown  no  
...        ...  ...   ...       ...       ...    ...       ...      ...  ..  
4516  cellular   30   jul       329         5     -1         0  unknown  no  
4517   unknown    9   may       153         1     -1         0  unknown  no  
4518  cellular   19   aug       151        11     -1         0  unknown  no  
4519  cellular    6   feb       129         4    211         3    other  no  
4520  cellular    3   apr       345         2    249         7    other  no  

[4521 rows x 17 columns]

#After that, I am taking only 10 observations(rows) by typing the my file wine_data and head(10). 10 in breckets because i want to take only 10 rows in order to not analyzing every 4521 person.

In [9]:
wine_data.head(10)

age            job  marital  education default  balance housing loan  \
0   30     unemployed  married    primary      no     1787      no   no   
1   33       services  married  secondary      no     4789     yes  yes   
2   35     management   single   tertiary      no     1350     yes   no   
3   30     management  married   tertiary      no     1476     yes  yes   
4   59    blue-collar  married  secondary      no        0     yes   no   
5   35     management   single   tertiary      no      747      no   no   
6   36  self-employed  married   tertiary      no      307     yes   no   
7   39     technician  married  secondary      no      147     yes   no   
8   41   entrepreneur  married   tertiary      no      221     yes   no   
9   43       services  married    primary      no      -88     yes  yes   

    contact  day month  duration  campaign  pdays  previous poutcome   y  
0  cellular   19   oct        79         1     -1         0  unknown  no  
1  cellular   11   may       220         1    339         4  failure  no  
2  cellular   16   apr       185         1    330         1  failure  no  
3   unknown    3   jun       199         4     -1         0  unknown  no  
4   unknown    5   may       226         1     -1         0  unknown  no  
5  cellular   23   feb       141         2    176         3  failure  no  
6  cellular   14   may       341         1    330         2    other  no  
7  cellular    6   may       151         2     -1         0  unknown  no  
8   unknown   14   may        57         2     -1         0  unknown  no  
9  cellular   17   apr       313         1    147         2  failure  no

#Then in order to take only mumbers, I am using 'pearson' method. 

In [5]:
wine_data.corr(method='pearson')

age   balance       day  duration  campaign     pdays  previous
age       1.000000  0.083820 -0.017853 -0.002367 -0.005148 -0.008894 -0.003511
balance   0.083820  1.000000 -0.008677 -0.015950 -0.009976  0.009437  0.026196
day      -0.017853 -0.008677  1.000000 -0.024629  0.160706 -0.094352 -0.059114
duration -0.002367 -0.015950 -0.024629  1.000000 -0.068382  0.010380  0.018080
campaign -0.005148 -0.009976  0.160706 -0.068382  1.000000 -0.093137 -0.067833
pdays    -0.008894  0.009437 -0.094352  0.010380 -0.093137  1.000000  0.577562
previous -0.003511  0.026196 -0.059114  0.018080 -0.067833  0.577562  1.000000

#Then I am using Kendal correlation method.
#Kendall correlation method is used to test similarities in the ordering of data when it is ranked by quantities,so,because of that i used Kendall method.
According to this method, only pdays (which means, the days that passed after contacting this client from a previous marketing phone call campaign) and previous (which means, number of contacts made before this campaign and for this client) are intersecting here saying that when pdays rise previous also rise.   





In [13]:
wine_data.corr(method='kendall')

age   balance       day  duration  campaign     pdays  previous
age       1.000000  0.051505 -0.017031 -0.024849  0.018897 -0.000840  0.000526
balance   0.051505  1.000000 -0.007962  0.035923 -0.024134  0.048403  0.054271
day      -0.017031 -0.007962  1.000000 -0.034289  0.108478 -0.066586 -0.062743
duration -0.024849  0.035923 -0.034289  1.000000 -0.066681  0.028551  0.031875
campaign  0.018897 -0.024134  0.108478 -0.066681  1.000000 -0.119233 -0.119460
pdays    -0.000840  0.048403 -0.066586  0.028551 -0.119233  1.000000  0.905505
previous  0.000526  0.054271 -0.062743  0.031875 -0.119460  0.905505  1.000000

#This is info about our file, where we can see that we have 4521 rows(obseervations or persons), 17 columns(features) one of them are objects like, job:services and some of them are int(numbers) like, age:35. dtypes(objects) shows that int is 64 which means numbers Integer (-9223372036854775808 to 9223372036854775807) and number of int64 is (7) as you can see: age, balance, day, duration, campaign, pdays, previous. Objects number is 10: job, marital, education default, housing, loan, contact, month, poutcome, y. And memory usage is 600,6+ KB(1KB = 1024 bytes).

In [16]:
wine_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4521 entries, 0 to 4520
Data columns (total 17 columns):
age          4521 non-null int64
job          4521 non-null object
marital      4521 non-null object
education    4521 non-null object
default      4521 non-null object
balance      4521 non-null int64
housing      4521 non-null object
loan         4521 non-null object
contact      4521 non-null object
day          4521 non-null int64
month        4521 non-null object
duration     4521 non-null int64
campaign     4521 non-null int64
pdays        4521 non-null int64
previous     4521 non-null int64
poutcome     4521 non-null object
y            4521 non-null object
dtypes: int64(7), object(10)
memory usage: 600.6+ KB


#Then we are using "describe" function in order to know the description of this file: 
#Count(Number of non-null observations)

In our case we have 4521 count non-null(having a value) observations that can be count.
#Mean(Mean or average of Values)

Mean is counted for every numerical feature and counted by adding all the numbers in age for example and by dividing them for total number of age.
#Std(standard deviation of the values, Standard Deviation we have a "standard" way of knowing what is normal, and what is extra large or extra small)

Every feature has their standard and its counted by following the wildspread formula of standard deviaton.
#Min(minimum value)

It is the minimum number of all values here 
#Percentages as 25%, 50%, 75% are given by default, 25%(formula = percentile * n (n is number of values))
#50%((formula = percentile * n (n is number of values))
#75%((formula = percentile * n (n is number of values))
#Max(maximum value) 

it is the maximim value of all numbers 





In [17]:
wine_data.describe()

age       balance          day     duration     campaign  \
count  4521.000000   4521.000000  4521.000000  4521.000000  4521.000000   
mean     41.170095   1422.657819    15.915284   263.961292     2.793630   
std      10.576211   3009.638142     8.247667   259.856633     3.109807   
min      19.000000  -3313.000000     1.000000     4.000000     1.000000   
25%      33.000000     69.000000     9.000000   104.000000     1.000000   
50%      39.000000    444.000000    16.000000   185.000000     2.000000   
75%      49.000000   1480.000000    21.000000   329.000000     3.000000   
max      87.000000  71188.000000    31.000000  3025.000000    50.000000   

             pdays     previous  
count  4521.000000  4521.000000  
mean     39.766645     0.542579  
std     100.121124     1.693562  
min      -1.000000     0.000000  
25%      -1.000000     0.000000  
50%      -1.000000     0.000000  
75%      -1.000000     0.000000  
max     871.000000    25.000000

#GoodBye!!!